MediaBrowser.Controller/Net/BasePeriodicWebSocketListener.cs

```diff
@@ -25,11 +26,19 @@ public abstract class BasePeriodicWebSocketListener<TReturnDataType, TStateType>
         where TStateType : WebSocketListenerState, new()
         where TReturnDataType : class
     {
+        private readonly Channel<bool> _channel = Channel.CreateUnbounded<bool>(new UnboundedChannelOptions
```

Comment:
Would it not be better to use a `BlockingCollection` with the messages directly?

Reply:
A blockingcollection will block a thread when there's no data. Channels use async-await. I don't think there's ever a good reason to use it.